src/EnergyPlus/UnitVentilator.cc

```diff
@@ -1789,165 +1425,133 @@ namespace UnitVentilator {
+        bool PrintFlag;   // TRUE when sizing information is reported in the eio file
+        int SAFMethod(0); // supply air flow rate sizing method (SupplyAirFlowRate, FlowPerFloorArea, FractionOfAutosizedCoolingAirflow,
+                          // FractionOfAutosizedHeatingAirflow ...)
+        int CapSizingMethod(0);    // capacity sizing methods (HeatingDesignCapacity, CapacityPerFloorArea, FractionOfAutosizedCoolingCapacity, and
+                                   // FractionOfAutosizedHeatingCapacity )
+        Real64 WaterCoilSizDeltaT; // water coil deltaT for design water flow rate autosizing
+
+        int PltSizHeatNum = 0;
+        bool ErrorsFound = false;
+        bool IsAutoSize = false;
+        Real64 MaxAirVolFlowDes = 0.0;
+        Real64 MaxAirVolFlowUser = 0.0;
+        Real64 OutAirVolFlowDes = 0.0;
+        Real64 OutAirVolFlowUser = 0.0;
+        Real64 MinOutAirVolFlowDes = 0.0;
+        Real64 MinOutAirVolFlowUser = 0.0;
+        Real64 MaxVolHotWaterFlowDes = 0.0;
+        Real64 MaxVolHotWaterFlowUser = 0.0;
+        Real64 MaxVolHotSteamFlowDes = 0.0;
+        Real64 MaxVolHotSteamFlowUser = 0.0;
+        Real64 MaxVolColdWaterFlowDes = 0.0;
+        Real64 MaxVolColdWaterFlowUser = 0.0;
+        Real64 CoolingAirVolFlowScalable = 0.0;
+        Real64 HeatingAirVolFlowScalable = 0.0;
         state.dataSize->DataScalableSizingON = false;
         state.dataSize->DataScalableCapSizingON = false;
-        CompType = state.dataUnitVentilators->cMO_UnitVentilator;
-        CompName = state.dataUnitVentilators->UnitVent(UnitVentNum).Name;
-        state.dataSize->DataZoneNumber = state.dataUnitVentilators->UnitVent(UnitVentNum).ZonePtr;
+        std::string CompType = state.dataUnitVentilators->cMO_UnitVentilator;
+        std::string CompName = unitVent.Name;
+        state.dataSize->DataZoneNumber = unitVent.ZonePtr;
         state.dataSize->ZoneCoolingOnlyFan = false;
         state.dataSize->ZoneHeatingOnlyFan = false;
-        DoWaterCoilSizing = false;
-        CoilNum = 0;
-        if (state.dataUnitVentilators->UnitVent(UnitVentNum).FanType_Num == DataHVACGlobals::FanType_SystemModelObject) {
+        bool DoWaterCoilSizing = false;
+        int CoilNum = 0;
+        if (unitVent.FanType_Num == DataHVACGlobals::FanType_SystemModelObject) {
             state.dataSize->DataFanEnumType = DataAirSystems::ObjectVectorOOFanSystemModel;
         } else {
             state.dataSize->DataFanEnumType = DataAirSystems::StructArrayLegacyFanModels;
         }
-        state.dataSize->DataFanIndex = state.dataUnitVentilators->UnitVent(UnitVentNum).Fan_Index;
+        state.dataSize->DataFanIndex = unitVent.Fan_Index;
         // unit ventilator is always blow thru
         state.dataSize->DataFanPlacement = DataSizing::ZoneFanPlacement::BlowThru;
 
-        if (state.dataUnitVentilators->UnitVent(UnitVentNum).CoilOption == BothOption) {
+        switch (unitVent.CoilOption) {
```

Comment:
Also not a big fan of using control flow (`if` or `switch`) to set boolean flags. Just set the boolean flags directly to the condition being tested by the control flow, e.g., 

```cpp
state.dataSize->ZoneCoolingOnlyFan = (unitVent.CoilOption == CoilOption::Both) || (unitVent.CoilOption == CoilOption::Cooling);
state.dataSize->ZoneHeatingOnlyFan = (unitVent.CoilOption == CoilOption::Both) || (unitVent.CoilOption == CoilOption::Heating);
```